src/lin.py

```diff
@@ -44,26 +44,30 @@ def strict_border_array(x: str) -> list[int]:
     return ba
 
 def kmp(x, p):
-    ba = strict_border_array(p)
+    if not x or not p:
+        return 
+    ba = border_array(p)
+    bax = strict_border_array(p)
```

Comment:
Do you need both of these?

Reply:
Vi oplever problemer med matchcasen, hvis vi også bruger bax der. 
F.eks. for p = "ccc".
Kan du se, hvad problemet er?

Reply:
Jeg ved ikke hvad problemet er. `bax` for `"ccc"` bør vel være `[0, 0, 2]` ikke?